src/ApplePay/Provider/OrderPaymentApplePayDirectProvider.php

```diff
@@ -34,12 +35,23 @@ final class OrderPaymentApplePayDirectProvider implements OrderPaymentApplePayDi
 {
     public function __construct(
         private readonly PaymentFactoryInterface $paymentFactory,
-        private readonly StateMachineFactoryInterface $stateMachineFactory,
+        private readonly FactoryInterface|StateMachineInterface $stateMachineFactory,
         private readonly RepositoryInterface $paymentMethodRepository,
         private readonly RepositoryInterface $gatewayConfigRepository,
         private readonly OrderCreationPaymentTokenProviderInterface|PaymentTokenProviderInterface $paymentTokenProvider,
         private readonly Payum $payum,
     ) {
+        if ($this->stateMachineFactory instanceof FactoryInterface) {
+            trigger_deprecation(
+                'sylius/mollie-plugin',
+                '2.1',
+                sprintf(
+                    'Passing an instance of "%s" as the second argument is deprecated. It will accept only instances of "%s" in Mollie 3.0.',
```

Comment:
```suggestion
                    'Passing an instance of "%s" as the second argument is deprecated. It will accept only instances of "%s" in MolliePlugin 3.0.',
```
🤔